include/functions.inc.php

```diff
@@ -716,7 +717,7 @@ function xhtml_cleanup($html) {
 function serendipity_fetchAuthor($author) {
     global $serendipity;
 
-    return serendipity_db_query("SELECT * FROM {$serendipity['dbPrefix']}authors WHERE " . (is_numeric($author) ? "authorid={$author};" : "username='" . serendipity_db_escape_string($author) . "';"));
+    return serendipity_db_query("SELECT * FROM {$serendipity['dbPrefix']}authors WHERE " . (is_numeric($author) ? "authorid={$author};" : "username='" . serendipity_db_escape_string($author) . "';"), true, 'assoc');
```

Comment:
Assuming the function was not broken before that looks like an API change we can't simply make. It might break plugins.